iModelCore/ecobjects/src/ECInstance.cpp

```diff
@@ -2687,14 +2687,22 @@ struct  InstanceXmlReader
                 return InstanceReadStatus::ECSchemaNotFound;
                 }
 
+            // do not search for the class if its schema is going to be pruned.

```

Comment:
This doesn't make sense to me.  If the schema has been pruned it won't be found so how would it even get to this code?

I suggest you write a test case where some schemas are pruned so you understand the problem before you write a solution.

Reply:
We already have a test for this scenario: SchemaDeserializationTest.PruneCAFromPrunedEC32Schemas.
In this test, we are pruning the referenced schema. The problem occurs because when ECInstanceReadContext::FindSchemaCP() can’t locate the schema, it returns a fallback schema, which, in this situation, ends up being the actual schema that relies on the reference schema.